File: pt-br/src/BuscaBinaria.ts
export default function BuscaBinaria(vetor: any[], alvo: number) {
  let ponteiroEsquerda = 0
  let ponteiroDireita = vetor.length - 1
  let indiceDoMeio = Math.floor(ponteiroEsquerda + (ponteiroDireita - ponteiroEsquerda) / 2)

  while (vetor.length > indiceDoMeio) {
    let numeroDoMeio = vetor[indiceDoMeio]

    if (numeroDoMeio === alvo) return indiceDoMeio

    if (numeroDoMeio > alvo) {
      ponteiroDireita = indiceDoMeio - 1
      BuscaBinaria(vetor[indiceDoMeio], alvo)
    }

    if (numeroDoMeio < alvo) {
      ponteiroEsquerda = indiceDoMeio + 1
      BuscaBinaria(vetor[indiceDoMeio], alvo)
    }

    if (ponteiroEsquerda > ponteiroDireita) throw new Error('[erro] índice não encontrado')

    indiceDoMeio = Math.floor(ponteiroEsquerda + (ponteiroDireita - ponteiroEsquerda) / 2)
  }
  return -1
}

File: pt-br/src/python/BuscaBinaria.py
def BuscaBinaria(vetor, alvo):
	ponteiroEsquerda = 0
	ponteiroDireita = len(vetor) - 1
	indiceDoMeio = 0

	while ponteiroEsquerda <= ponteiroDireita:

		indiceDoMeio = ponteiroEsquerda + (ponteiroDireita - ponteiroEsquerda) // 2

		if vetor[indiceDoMeio] < alvo:
			ponteiroEsquerda = indiceDoMeio + 1

		elif vetor[indiceDoMeio] > alvo:
			ponteiroDireita = indiceDoMeio - 1

		else:
			return indiceDoMeio
	return -1
